src/libsystemd/sd-event/event-source.h

```diff
@@ -44,6 +45,19 @@ typedef enum WakeupType {
 
 struct inode_data;
 
+struct ratelimit_state {
```

Comment:
maybe call this "ratelimit_saved_state", since this is purely about saving/restoring stuff